library/core/src/panic/location.rs

```diff
@@ -28,8 +30,21 @@ use crate::fmt;
 /// Files are compared as strings, not `Path`, which could be unexpected.
 /// See [`Location::file`]'s documentation for more discussion.
 #[lang = "panic_location"]
+#[derive(Copy, Clone)]
+#[stable(feature = "panic_hooks", since = "1.10.0")]
+#[cfg(not(bootstrap))]
+pub struct Location<'a> {
+    file: *const (),
+    _file_actual: PhantomData<&'a CStr>,
```

Comment:
Why does `Location` have a lifetime? Isn't it always `'static`?

Reply:
I don't know either - but it's stable :(